src/psyclone/psyir/nodes/loop.py

```diff
@@ -59,8 +59,10 @@ class Loop(Statement):
 
     :param parent: parent of this node in the PSyIR.
     :type parent: sub-class of :py:class:`psyclone.psyir.nodes.Node`
-    :param str variable_name: optional name of the loop iterator \
-        variable. Defaults to an empty string.
+    :param variable: optional reference to the loop iterator \
+        variable. Defaults to None.
```

Comment:
What does giving variable a None value mean here? That it is just not complete and at some point it will need to have the variable setter give a symbol before reaching the back-end, or that it is an unamed/independent variable and if the backend find a None they can create/reuse an independent variable here.

It is OK if it doesn't happen yet, I am just asking whats your view here.

Reply:
It is there to mean that it is not complete. It allows a loop to be created before the name is known if that is required. To better reflect this I have made the `_check_variable()` method fail if the variable is `None` and no longer call `_check_variable()` in the constructor if its value is `None`. This means that the `create` and `setter` methods must now both supply a valid DataSymbol instance. Rightly or wrongly it does mean that the backends will need to check that a value has been provided.